### PR TITLE
Faulty evidence error handling

### PR DESCRIPTION
**Pull request by setrofim**
Ensure correct handing of various errors in the provided evidence during verification. The general idea is that problems with the service setup should be communicted as errors (ultimately resulting in a 500 response from the REST frontend), while problems with the evidence should be communicated via an appropriately composed attestation result (with in a 200 response from the frontend).

This is done by creating an error type specifically to indicate bad evidnece, and handling it differently from others. Liminations in how errors can be represented in AR4SI are mitegated by providing additional information a a "problem" policy claim.

### CODE REVIEW

**Comment by setrofim** [comment]
This addresses https://github.com/veraison/services/issues/107 and  https://github.com/veraison/services/issues/109

File: integration-tests/tests/test_end_to_end.tavern.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ marks:
       vals:
         - [psa, p1, good, full, ec.p256, good]
         - [psa, p1, good, mini, ec.p256, good]
+        - [psa, p1, missingclaims, mini, ec.p256, noident]
```
Comment (thomas-fossati) [inline integration-tests/tests/test_end_to_end.tavern.yaml:26]: Why is this mapped to 99?  Is the signature incorrect?

File: integration-tests/tests/test_end_to_end.tavern.yaml [hunk 2]
```diff
@@ -23,6 +23,9 @@ marks:
...
+        - [psa, p1, good, mini, bad, badcrypto]
+        - [psa, p1, badinstance, full, ec.p256, badinstance]
```
Comment (thomas-fossati) [inline integration-tests/tests/test_end_to_end.tavern.yaml:28]: ditto: why 99? If the identity is not known (i.e., the trust anchor lookup fails) shouldn't we return instance-identity=1 or 97?

File: integration-tests/tests/test_end_to_end.tavern.yaml (hunk 1 above)
Comment (thomas-fossati) [reply integration-tests/tests/test_end_to_end.tavern.yaml:26]: as per f2f discussion, this needs a new AR4SI code point.

**Review by thomas-fossati — APPROVED** [review]
LGTM

**Pull request merged**